src/ValidationUnit.js

```diff
@@ -185,6 +185,10 @@ export default class ValidationUnit {
     return this.setRequirement(requiredFunc, message, 'isRequired');
   }
 
+  isRequiredWhen(func, message = '{name} is required.') {
+    return this.setRequirement((val, allValues) => func(val, allValues) && requiredFunc(val, allValues) || !func(val, allValues), message, 'isRequired');
```

Comment:
&& has a higher precedence than || so it's `(func(val, allValues) && requiredFunc(val, allValues)) || !func(val, allValues)`.  But I agree that with the grouping it is a lot easier to reason about quickly and makes the intention clear.  Added!